functional_tests/histogram/histogram_test.go

```diff
@@ -166,13 +166,14 @@ func runMetricsTest(t *testing.T, isHistogram bool, metricsSink *consumertest.Me
 	if os.Getenv("GENERATE_EXPECTED") == "true" {
 		outputDir := filepath.Join("testdata", "expected", majorMinor)
 		require.NoError(t, os.MkdirAll(outputDir, 0o755))
+		require.NotNil(t, actualMetrics, "Did not receive any metrics for component %s", input.ServiceName)
 		internal.ReduceDatapoints(actualMetrics, 1)
 		err := golden.WriteMetrics(t, filepath.Join(outputDir, fileName), *actualMetrics)
 		require.NoError(t, err)
 	}
 
-	assert.NotNil(t, actualMetrics, "Did not receive any metrics for component %s", input.ServiceName)
 	internal.MaybeUpdateExpectedMetricsResults(t, filepath.Join(testDir, fileName), actualMetrics)
+	require.NotNil(t, actualMetrics, "Did not receive any metrics for component %s", input.ServiceName)
```

Comment:
should this go before line 175 since we have nothing to write if actual metrics are nil

Reply:
it doesn't crash if it is not updating, but it won't be a problem if swapped